### PR TITLE
fix service list can not search by groupName only

### LINKED ISSUE #4282: 1.4.0 Service list Cannot query by service name
**Describe the bug**
image
image

### PR DESCRIPTION
**Pull request by mark4z**
Fixes #4282

### CODE REVIEW

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java [hunk 1]
```diff
@@ -73,4 +73,9 @@ public static void checkServiceNameFormat(String combineServiceName) {
                     "Param 'serviceName' is illegal, it should be format as 'groupName@@serviceName'");
         }
     }
+    
+    public static String getGroupedNameOptional(final String serviceName, final String groupName) {
```
Comment (sanxun0325) [inline api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:77]: You can add some comments

Comment (mark4z) [reply api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:77]: Ok,fine

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java [hunk 2]
```diff
@@ -27,6 +27,20 @@
...
+     *
+     * In most cases, serviceName can not be nil. In other cases, for search or anything,
+     * See {@link com.alibaba.nacos.api.naming.utils.NamingUtils#getGroupedNameOptional(String, String)}
+     *
+     * etc: serviceName | groupName | result
+     *      serviceA    | groupA    | serviceA@@groupA
+     *      nil         | groupA    | threw IllegalArgumentException
+     *
+     *
+     * @return 'serviceName@@groupName'
```
Comment (KomachiSion) [inline api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:42]: serviceName@@groupName ? should groupName@@serviceName I think

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java [hunk 3]
```diff
@@ -27,6 +27,20 @@
...
     
+    /**
+     * Returns a combined string with serviceName and groupName.
+     * serviceName can not be nil.
+     *
+     * In most cases, serviceName can not be nil. In other cases, for search or anything,
+     * See {@link com.alibaba.nacos.api.naming.utils.NamingUtils#getGroupedNameOptional(String, String)}
+     *
+     * etc: serviceName | groupName | result
+     *      serviceA    | groupA    | serviceA@@groupA
```
Comment (KomachiSion) [inline api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:38]: serviceA@@groupA --》 groupA@@serviceA

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java [hunk 4]
```diff
@@ -73,4 +87,23 @@ public static void checkServiceNameFormat(String combineServiceName) {
...
+     *
+     * This method works similar with {@link com.alibaba.nacos.api.naming.utils.NamingUtils#getGroupedName}
+     * But not verify any parameters.
+     *
+     * etc: serviceName | groupName | result
+     *      serviceA    | groupA    | serviceA@@groupA
+     *      nil         | groupA    | @@groupA
+     *      nil         | nil       | @@
+     *
+     * @return 'serviceName@@groupName'
```
Comment (KomachiSion) [inline api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:103]: same problem with above

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java (hunk 3 above)
Comment (mark4z) [reply api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java]: Sorry, i have fix it.

**Comment by KomachiSion** [comment]
Execute me, Your commit log has error. I think you need to fix it By rebase or checkout a new branch to modify and submit PR again.

**Comment by KomachiSion** [comment]
image

**Pull request closed without merging**